libs/domain/user/registration/registration.component.ts

```diff
@@ -0,0 +1,217 @@
+import { when } from 'lit/directives/when.js';
+import { firstValueFrom } from 'rxjs';
+import { RegistrationOptions } from './registration.model';
+import { styles } from './registration.styles';
+
+@defaultOptions({
+  passwordVisibility: PasswordVisibilityStrategy.Click,
+})
+@hydrate({ event: ['mouseover', 'focus'] })
+export class UserRegistrationComponent extends ContentMixin<RegistrationOptions>(
+  LitElement
+) {
+  static styles = styles;
+
+  @state() isLoading?: boolean = false;
+  @state() hasGenericError?: boolean = false;
+  @state() hasAgreementError?: boolean = false;
+  @state() hasPasswordError?: boolean = false;
+
+  @query('form') form?: HTMLFormElement;
+  @query('oryx-password-input') passwordComponent?: PasswordInputComponent;
+
+  @query('select[name=salutation]') salutation?: HTMLSelectElement;
+  @query('select[name=gender]') gender?: HTMLSelectElement;
+  @query('input[name=firstName]') firstName?: HTMLInputElement;
+  @query('input[name=lastName]') secondName?: HTMLInputElement;
+  @query('input[name=email]') email?: HTMLInputElement;
+  @query('input[name=password]') password?: HTMLInputElement;
+  @query('input[name=acceptedTerms]') acceptedTerms?: HTMLInputElement;
+
+  protected fieldRenderer = resolve(FormRenderer);
+  protected registrationService = resolve(RegistrationService);
+
+  protected salutationService = resolve(SalutationService);
+  protected salutations = signal(this.salutationService.get());
+
+  protected genderService = resolve(GenderService);
+  protected genders = signal(this.genderService.get());
+
+  protected onSubmit(event: Event): void {
+    event.preventDefault();
+
+    if (this.validateForm()) {
+      this.registerUser();
+    }
+  }
+
+  protected validateForm(): boolean {
+    this.setCheckboxValidity();
+    this.setPasswordValidity();
+    return !!(
+      this.form?.checkValidity() &&
+      this.acceptedTerms?.checked &&
+      this.passwordComponent?.isValid()
+    );
+  }
+
+  protected setCheckboxValidity(): void {
+    this.hasAgreementError = !this.acceptedTerms?.checked;
+  }
+
+  protected setPasswordValidity(): void {
+    this.hasPasswordError = !this.passwordComponent?.isValid();
+  }
+
+  protected async registerUser(): Promise<void> {
+    this.isLoading = true;
+
+    try {
+      await firstValueFrom(
+        this.registrationService.register(this.getFormData())
+      );
+      this.hasGenericError = false;
+    } catch (e) {
+      this.hasGenericError = true;
+    } finally {
+      this.isLoading = false;
+    }
+  }
+
+  protected override render(): TemplateResult {
+    return html`
+      ${when(
+        this.hasGenericError,
+        () => html`
+          <oryx-notification type="error" scheme="dark">
+            ${this.i18n('user.registration.something-went-wrong')}.
+          </oryx-notification>
+        `
+      )}
+
+      <form @submit=${this.onSubmit}>
+        ${this.fieldRenderer.buildForm(this.getFields())}
```

Comment:
you could actually register the password and terms and conditions as form field renderers. That would keep this class much cleaner, as it's a bit lengthy. 

It means that all form fields are data driven, and when password or t&c is rendered, your registered Form field renderer would be used.